runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -842,9 +864,34 @@ J9::TransformUtil::transformIndirectLoad(TR::Compilation *comp, TR::Node *node)
                }
             else if (symrefIsImprovable)
                {
-               uintptrj_t targetObjectReference;
-               TR::SymbolReference *improvedSymRef;
+               uintptrj_t targetObjectReference = 0;
+               TR::SymbolReference *improvedSymRef = node->getSymbolReference();
 
+#if defined(JITSERVER_SUPPORT)
+               if (comp->isOutOfProcessCompilation())
+                  {
+                  TR::KnownObjectTable *knot = comp->getOrCreateKnownObjectTable();
+                  bool knotEnabled = (knot != NULL);
+                  auto stream = TR::CompilationInfo::getStream();
+                  stream->write(JITServer::MessageType::KnownObjectTable_getReferenceField,
+                        baseObject->getSymbol()->isStatic(), baseObjectRefLocation, fieldOffset, knotEnabled);
+
+                  auto recv = stream->read<TR::KnownObjectTable::Index, uintptrj_t*, uintptrj_t>();
+                  TR::KnownObjectTable::Index knotIndex = std::get<0>(recv);
+                  uintptrj_t *objectPointerReference = std::get<1>(recv);
+                  targetObjectReference = std::get<2>(recv);
```

Comment:
`targetObjectReference` is not needed

Reply:
`targetObjectReference` is used afterwards at https://github.com/eclipse/openj9/blob/fd96afe5f58b9125d812dcb45ae85f7376e0b7d3/runtime/compiler/optimizer/J9TransformUtil.cpp#L862-L872.